env/env.go

```diff
@@ -0,0 +1,141 @@
+package env
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strconv"
+)
+
+// Lookup must return the value for the given name and whether or not it was
+// found.
+type Lookup func(name string) (string, bool)
+
+// VarSet provides methods to access environment variables from a list of Lookup
+// sources.
+type VarSet struct {
+	// sources are searched sequentially for a given environment variable.
+	sources []Lookup
+}
+
+// NewVarSet makes a new VarSet from opts.
+func NewVarSet(opts ...VarSetOpt) *VarSet {
+	v := &VarSet{sources: make([]Lookup, 0)}
+	for _, opt := range opts {
+		opt(v)
+	}
+	return v
+}
+
+// AddSource adds s as another lookup source.
+func (v *VarSet) AddSource(s Lookup) *VarSet {
+	v.sources = append(v.sources, s)
+	return v
+}
+
+// String gets the string environment variable with the given name, if set.
+// If not found, returns false.
```

Comment:
Incorrect comment - if not found, returns `defaultVal`

Reply:
Ooops.